lib/rules/declaration-block-no-redundant-longhand-properties/index.mjs

```diff
@@ -284,7 +284,14 @@ const rule = (primary, secondaryOptions, context) => {
 						longhandSubPropertiesOfShorthandProperties.get(shorthandProperty),
 					);
 
-					ignoreLonghands.forEach((value) => shorthandProps.delete(value));
+					// see 9ab4fb2 for the historical reason behind the exception
+					const isBorder = shorthandProperty === 'border';
+
+					shorthandProps.forEach((value) => {
+						const mustDiscard = isBorder && value.length !== 12;
+
+						if (ignoreLonghands.includes(value) || mustDiscard) shorthandProps.delete(value);
+					});
```

Comment:
[suggest] I think we can just customize `longhandSubPropertiesOfShorthandProperties` for this rule like this:

```js
const shorthandToLonghands = new Map(longhandSubPropertiesOfShorthandProperties);
shorthandToLonghands.set('border', new Set(['border-width', 'border-style', 'border-color']));
```

I'd like to avoid the magic number `12` (in `value.length !== 12`).

Reply:
I thought about it, I chose not to. I hesitated because this change is hacky and preferred the diff to be small.
Ill add a `const`.

Reply:
Even after extracting the magic number to a local variable via 63b41d7, the number is still present. This means that we have to fix the hack later.

I think we should write non-hack code if possible.

Reply:
I meant that not fixing `properties.mjs` is sloppy and I think that this exception will be temporary.
i.e. `border` should be updated in the next major

Reply:
I suggest adding the following code to `lib/rules/declaration-block-no-redundant-longhand-properties/index.mjs`, not `properties.mjs`. E.g.

```diff
const meta = {
	url: 'https://stylelint.io/user-guide/rules/declaration-block-no-redundant-longhand-properties',
	fixable: true,
};

+const shorthandToLonghands = new Map(longhandSubPropertiesOfShorthandProperties);
+shorthandToLonghands.set('border', new Set(['border-width', 'border-style', 'border-color']));
```

This code doesn't mutate the `longhandSubPropertiesOfShorthandProperties` object, so there is no effect on other modules.

Reply:
Thanks for the review but I realised that I was wrong along the way.
So this is moot :)